controllers/nutanixmachine_controller.go

```diff
@@ -327,81 +311,109 @@ func (r *NutanixMachineReconciler) reconcileDelete(rctx *nctx.MachineContext) (r
 		log.Error(errorMsg, "failed to delete VM")
 		return reconcile.Result{}, errorMsg
 	}
+
 	// Check if VMUUID is absent
 	if vmUUID == "" {
 		log.Info(fmt.Sprintf("VMUUID was not found in spec for VM %s. Skipping delete", vmName))
-	} else {
-		// Search for VM by UUID
-		vm, err := FindVMByUUID(ctx, v3Client, vmUUID)
-		// Error while finding VM
+		log.Info(fmt.Sprintf("Removing finalizers for VM %s during delete reconciliation", vmName))
+		ctrlutil.RemoveFinalizer(rctx.NutanixMachine, infrav1.NutanixMachineFinalizer)
+		return reconcile.Result{}, nil
+	}
+
+	// Search for VM by UUID
+	vm, err := FindVMByUUID(ctx, v3Client, vmUUID)
+	// Error while finding VM
+	if err != nil {
+		errorMsg := fmt.Errorf("error finding vm %s with uuid %s: %v", vmName, vmUUID, err)
+		log.Error(errorMsg, "error finding vm")
+		conditions.MarkFalse(rctx.NutanixMachine, infrav1.VMProvisionedCondition, infrav1.DeletionFailed, capiv1.ConditionSeverityWarning, errorMsg.Error())
+		return reconcile.Result{}, errorMsg
+	}
+
+	// Vm not found
```

Comment:
nit: lets be consistent with "VM" or "vm" in all the comments and info and error messages
```suggestion
	// VM not found
```